addons/statemachine/fnc_addTransition.sqf

```diff
@@ -49,6 +50,30 @@ if (isNull _stateMachine
     || {_condition isEqualTo {}}
 ) exitWith {false};
 
+if (_conditionFrequency > 0) then {
+    _condition = compile format [QUOTE( \
+        true && \
```

Comment:
?

Reply:
condition needs a boolean before any codeblock/function right?

Reply:
```sqf
true &&\
        {
```
is literally the same as
```sqf
call {
```
Question is why you need the `call` in the first place.

Reply:
```sqf
if (_conditionFrequency > 0) then {
    _condition = compile format [QUOTE(\
        private _return = false;\
        if ((_current getVariable [ARR_2((QQGVAR(lastCheckedState) + str _id), '')]) isEqualTo _thisState) then {\
            private _lastCheckedTime = _current getVariable [ARR_2((QQGVAR(lastCheckedTime) + str _id), CBA_MissionTime)];\
            if (CBA_MissionTime >= (_lastCheckedTime + %2)) then {\
                _current setVariable [ARR_2((QQGVAR(lastCheckedTime) + str _id), CBA_MissionTime)];\
                _return = true;\
            } else {\
                _return = false;\
            };\
        } else {\
            _current setVariable [ARR_2((QQGVAR(lastCheckedState) + str _id), _thisState)];\
            _current setVariable [ARR_2((QQGVAR(lastCheckedTime) + str _id), CBA_MissionTime)];\
            _return = false;\
        };\
        _return && %1\
    ), _condition, _conditionFrequency];
};
```

removed bool and {}, just _return bool && original condition